### PR TITLE
Remove pinnng scrollTop to 0

### PR DESCRIPTION
**Pull request by ofersadgat**
This is done so that if there are elements which appear above the current element, scrollOffset.top - this._positionFromTop will return a negative number thereby correctly telling the child that it is not being rendered. This allows the child to render only when it becomes necessary to do so as opposed to rendering always.

Thanks for contributing to react-virtualized!

**Before submitting a pull request,** please complete the following checklist:

- [x] The existing test suites (`npm test`) all pass
- [x] For any new features or bug fixes, both positive and negative test cases have been added
- [ ] For any new features, documentation has been added
- [x] For any documentation changes, the text has been proofread and is clear to both experienced users and beginners.
- [x] Format your code with [prettier](https://github.com/prettier/prettier) (`npm run prettier`).
- [x] Run the [Flow](https://flowtype.org/) typechecks (`npm run typecheck`).

Can add a test later if you feel necessary. I first want to see if there is a reason that this shouldnt be done.

### CODE REVIEW

**Comment by aem** [comment]
just got around to investigating this a bit. internally at my company we take advantage of the scroll pinning (do some subtraction, pass in the result, and if it's <= 0 then scroll to top) and i imagine we're not alone. i think a better approach would be to do this client-side, conditionally rendering the components based on screen position. seems like rendering/not rendering the element is a bit out of scope

**Comment by aem** [comment]
@wuweiweiwu would be interested to hear your thoughts

**Comment by ofersadgat** [comment]
@aem I confused, you say if the result is less than or equal to zero, so allowing negative numbers would still be handled correctly in your case, right?

**Comment by wuweiweiwu** [comment]
I think its reasonable for the render prop to pass in a negative value so that children can know that its above the virtualization window and perhaps return some sort of placeholder. 

I'll have too test this locally . @ofersadgat do you mind adding some tests? Theres a good chance this would be merged

**Comment by ofersadgat** [comment]
Yup, can do. I'm on vacation right now, so I'll submit it first thing when I get back (about 2 weeks).

**Comment by ofersadgat** [comment]
@wuweiweiwu to give you a heads up, I started taking a look at this again and found a few bugs: scrollTop is not updated if the position of the element updates (but because the position of the elements updates the scrollTop should update as well) so this will be a slightly larger change than anticipated. Please let me know if you see any problems with updating scrollTop in updatePosition. Thanks.

**Comment by wuweiweiwu** [comment]
@ofersadgat No problem! I will have to look at the specific changes in `updatePosition` but I think it should be fine

File: source/WindowScroller/WindowScroller.js [hunk 1]
```diff
@@ -91,9 +91,9 @@ export default class WindowScroller extends React.PureComponent<Props, State> {
 
   state = {
     ...getDimensions(this.props.scrollElement, this.props),
-    isScrolling: false,
     scrollLeft: 0,
     scrollTop: 0,
+    isScrolling: false,
```
Comment (ofersadgat) [inline source/WindowScroller/WindowScroller.js]: will move this back

**Comment by ofersadgat** [comment]
I want to write a test along the lines of:

```js
  it('should update scrollTop when a resize event occurs', () => {
    const renderFn = jest.fn();

    // Simulate scrolled documentElement
    const component = render(
      getMarkup({
        headerElements: <div id="a" style={{height: 501}} />,
        renderFn,
      }),
    );
    render._mountNode.querySelector('#a').size.height = '200px';
    expect(renderFn).lastCalledWith(
      expect.objectContaining({
        scrollTop: -200,
      }),
    );
  });
```
I.e. if a node above the current one changes size, that will cause a resize on the scollelement which should cause new props to get passed down. ut I'm not quite sure how to get this test to work with JSDom. Any help here would be appreciated.

Thanks!

**Comment by wuweiweiwu** [comment]
@ofersadgat hmm. if an element changes size, it will not trigger a rerender unless `forceUpdate` is called since these are all `PureComponents` 

This will take a bit more work then I initially thought. I'm not sure if this will introduce new bugs. 

Right now the changes look/sound fine, but we'll have to do some extensive smoke tests to make sure it doesnt introduce new bugs.

**Comment by ofersadgat** [comment]
@wuweiweiwu I'm confused. If the parent element changes size, then the size of the scrollElement will change which means that `_registerResizeListener` will call `_onResize` which will call `updatePosition` which will call setState with the new parameters. If the parameters have changed, we should get a rerender which will pass the new props down. Or have I missed something?

**Pull request closed without merging**